### PR TITLE
Small improvements to enemy control

### PR DESCRIPTION
**Pull request by xAstroBoy**
Expose RigidbodyController isSprinting and add isMoving to sync with enemy movement mechanisms.
@winstxnhdw as your requested

### CODE REVIEW

File: lc-hax/Scripts/Modules/Possession/Controllers/BaboonHawkController.cs [hunk 1]
```diff
@@ -1,5 +1,10 @@
 using Unity.Netcode;
 using Hax;
+enum Baboon {
+    Scouting = 0,
+    ReturningToCamp = 1,
+    FocusingOnThreat = 2,
+}
```
Comment (winstxnhdw) [inline lc-hax/Scripts/Modules/Possession/Controllers/BaboonHawkController.cs:7]: Will you be using this?

File: lc-hax/Scripts/Modules/Possession/Controllers/BrackenController.cs [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
     internal void UsePrimarySkill(FlowermanAI enemyInstance) {
-        if (!enemyInstance.carryingPlayerBody) return;
-
-        _ = enemyInstance.Reflect().InvokeInternalMethod("DropPlayerBody");
+        if (!enemyInstance.carryingPlayerBody) {
+            enemyInstance.SetBehaviourState(FlowerMan.ANGER);
+        }
+        else
+            _ = enemyInstance.Reflect().InvokeInternalMethod("DropPlayerBody");
         enemyInstance.DropPlayerBodyServerRpc();
```
Comment (winstxnhdw) [inline lc-hax/Scripts/Modules/Possession/Controllers/BrackenController.cs:17]: Why are we dropping the body twice?

Comment (xAstroBoy) [reply lc-hax/Scripts/Modules/Possession/Controllers/BrackenController.cs:17]: That’s the client version, to both force the bracken animation to do it

Comment (winstxnhdw) [reply lc-hax/Scripts/Modules/Possession/Controllers/BrackenController.cs:17]: Don't we only need one? Since the server RPC will call the client RPC?

Comment (winstxnhdw) [reply lc-hax/Scripts/Modules/Possession/Controllers/BrackenController.cs:17]:
```cs
    [ServerRpc]
    public void DropPlayerBodyServerRpc()
    {
        NetworkManager networkManager = base.NetworkManager;
        if ((object)networkManager == null || !networkManager.IsListening)
        {
            return;
        }

        if (__rpc_exec_stage != __RpcExecStage.Server && (networkManager.IsClient || networkManager.IsHost))
        {
            if (base.OwnerClientId != networkManager.LocalClientId)
            {
                if (networkManager.LogLevel <= LogLevel.Normal)
                {
                    Debug.LogError("Only the owner can invoke a ServerRpc that requires ownership!");
                }

                return;
            }

            ServerRpcParams serverRpcParams = default(ServerRpcParams);
            FastBufferWriter bufferWriter = __beginSendServerRpc(2817453984u, serverRpcParams, RpcDelivery.Reliable);
            __endSendServerRpc(ref bufferWriter, 2817453984u, serverRpcParams, RpcDelivery.Reliable);
        }

        if (__rpc_exec_stage == __RpcExecStage.Server && (networkManager.IsServer || networkManager.IsHost))
        {
            DropPlayerBodyClientRpc();
        }
    }

    [ClientRpc]
    public void DropPlayerBodyClientRpc()
    {
        NetworkManager networkManager = base.NetworkManager;
        if ((object)networkManager != null && networkManager.IsListening)
        {
            if (__rpc_exec_stage != __RpcExecStage.Client && (networkManager.IsServer || networkManager.IsHost))
            {
                ClientRpcParams clientRpcParams = default(ClientRpcParams);
                FastBufferWriter bufferWriter = __beginSendClientRpc(1942952026u, clientRpcParams, RpcDelivery.Reliable);
                __endSendClientRpc(ref bufferWriter, 1942952026u, clientRpcParams, RpcDelivery.Reliable);
            }

            if (__rpc_exec_stage == __RpcExecStage.Client && (networkManager.IsClient || networkManager.IsHost))
            {
                DropPlayerBody();
            }
        }
    }
```

**Review by winstxnhdw — APPROVED** [review]
LGTM

File: lc-hax/Scripts/Modules/Possession/Controllers/BrackenController.cs (hunk 2 above)
Comment (xAstroBoy) [reply lc-hax/Scripts/Modules/Possession/Controllers/BrackenController.cs]: Yeah no need, i dont know why controlcompany does that lol

**Comment by xAstroBoy** [comment]
All good

**Comment by winstxnhdw** [comment]
Apparently, in C#, implemented methods cannot be `internal`. I have to set it back to `public`.

**Pull request merged**